src/ILCompiler/src/Program.cs

```diff
@@ -491,7 +491,16 @@ private int Run(string[] args)
 
             bool supportsReflection = !_isReadyToRunCodeGen && !_isWasmCodegen && !_isCppCodegen && _systemModuleName == DefaultSystemModule;
 
-            MetadataManager compilationMetadataManager = supportsReflection ? metadataManager : (MetadataManager)new EmptyMetadataManager(typeSystemContext);
+            MetadataManager compilationMetadataManager;
+            if (_isReadyToRunCodeGen)
+            {
+                compilationMetadataManager = new ReadyToRunTableManager(typeSystemContext);
+            }
+            else
+            {
+                compilationMetadataManager = supportsReflection ? metadataManager : (MetadataManager)new EmptyMetadataManager(typeSystemContext);
```

Comment:
Non-blocking nit - I think it might be slightly easier to read to structure this as an if () - else if () - else block but I leave it to your styling preference :-).

Reply:
I think the ship has sailed on ternary operators in this file

https://github.com/dotnet/corert/blob/d55ebfd33d4d45afbdf1a3adedc32a839c5a4a69/src/ILCompiler/src/Program.cs#L476-L477

Reply:
No worries, I can easily live with that.